earth_enterprise/src/fusion/autoingest/AssetFactory.h

```diff
@@ -256,6 +257,199 @@ namespace AssetFactory
   }
 
 
+  template <class MutableDerivedVersionHandleType, class Version,
+            class ConfigType, class VersionDType, class Extras = int>
+  MutableDerivedVersionHandleType ReuseOrMakeAndUpdate(const std::string& ref_,
+                                                       AssetDefs::Type type_,
+                                                       const khMetaData& meta_,
+                                                       const ConfigType& config_)
+  {
+      using AssetHandleType = typename MutableDerivedVersionHandleType::Impl::MutableAssetType;
+      std::vector<SharedString> inputarg, boundInputs;
+      std::transform(inputarg.begin(), inputarg.end(), back_inserter(boundInputs),
+                     ptr_fun(&AssetVersionRef::Bind));
+
+      auto asset = Find<AssetHandleType>(ref_, type_);
+      if (asset)
+      {
+          for (const auto& v : asset->versions)
+          {
+              try
+              {
+                  VersionDType version(v);
+                  version.LoadAsTemporary();
+                  if ((version->state != AssetDefs::Offline) &&
+                     (version->inputs == boundInputs) &&
+                      config_.IsUpToDate(version->config))
+                  {
+                      version.MakePermanent();
+                      return version;
+                  }
+                  else
+                  {
+                      version.NoLongerNeeded();
+                  }
+              }
+              catch (...)
+              {
+                  notify(NFY_WARN,"could not reuse a version");
```

Comment:
I checked unit test coverage and this line is never hit. I'm not sure how much effort it's worth.